### PR TITLE
Add `thingId` validator

### PR DESCRIPTION
**Pull request by dwick**
A `thingId` is a type prefix: `tN` and a base36 number separated by an underscore.

`t3_1` is a valid `thingId` that previously didn't meet the length requirement.

:eyeglasses: @ajacksified

### CODE REVIEW

**Comment by ajacksified** [comment]
tests pls https://github.com/reddit/snoode/tree/master/test

**Comment by dwick** [comment]
after lunch!

**Comment by dwick** [comment]
:haircut: tests added

File: src/models/account.es6.js [hunk 1]
```diff
@@ -4,6 +4,16 @@ class Account extends Base {
...
     props._type = 'Account';
     super(props);
+
+    var account = this;
+
+    this.validators = {
+      thingId: function() {
+        var thingId = this.get('thingId');
+
+        return Base.validators.thingId(thingId, '2');
```
Comment (ajacksified) [inline src/models/account.es6.js]: `2` might be better off as a constant on the class (here and below models)

Comment (ajacksified) [inline src/models/account.es6.js]: The `thingId` validator should also be optional - it should only test the regex _if_ it is not undefined, such as when creating a new Account to be `POST`ed.

**Comment by ajacksified** [comment]
:nail_care:

**Comment by dwick** [comment]
:haircut: didn't add the `const`s as class members because i didnt see a use case, but they're at least defined as private `const`s now.

**Comment by dwick** [comment]
:haircut: removed specific type prefixes since they can be different per install.

**Comment by ajacksified** [comment]
:fish:

**Comment by dwick** [comment]
da68efe49f2f4c5ee45c547913ac375f3c9ff668

**Pull request closed without merging**